app/flatpak-builtins-repo.c

```diff
@@ -162,7 +200,7 @@ print_metadata (GVariant   *meta,
       while (g_variant_iter_next (&iter, "{&s(tt&s)}", &ref, &installed_size, &download_size, &metadata))
         {
           if (strcmp (branch, ref) == 0)
-            g_print ("%s", metadata);
+            g_print ("%s\n", metadata);
```

Comment:
Most metadata files have an trailing newline already, did you find one without it?

Reply:
yes, without this, I got the shell prompt tucked at the end of the last line, in my tests

Reply:
Here is an example:

> $ ./flatpak repo --metadata app/org.gnome.font-viewer/x86_64/master ~/Sources/gnome-font-viewer/repo
> [Application]
> name=org.gnome.font-viewer
> runtime=org.gnome.Platform/x86_64/master
> sdk=org.gnome.Sdk/x86_64/master
> command=gnome-font-viewer
> 
> [Context]
> shared=ipc;
> sockets=x11;wayland;fallback-x11;
> devices=dri;
> filesystems=xdg-run/dconf;~/.config/dconf:ro;
> 
> [Session Bus Policy]
> org.gtk.vfs.*=talk
> org.gtk.vfs=talk
> ca.desrt.conf=talk
> 
> [Environment]
> DCONF_USER_CONFIG_DIR=.config/dconf
> 
> [Extension org.gnome.font_viewer.Locale]
> directory=share/runtime/locale
> autodelete=true
> locale-subset=true
> 
> [Extension org.gnome.font_viewer.Debug]
> directory=lib/debug
> autodelete=true
> no-autodownload=true
> 
> [Build]
> built-extensions=org.gnome.font_viewer.Locale;org.gnome.font_viewer.Debug;[mclasen@localhost flatpak]$